### ISSUE TITLE
Gulp Watch not terminated on quit

### ISSUE THREAD

**Issue by jvskelton** [driftyco/ionic-cli#570]
This has been brought up in the past  on issue [#186] (https://github.com/driftyco/ionic-cli/issues/186).

So essentially, whenever you run "ionic serve"  followed by a "quit/q" or "Ctrl c" the gulp watch process does not actually terminate in windows.

There seems to be an orphaned node process that was spawned by "cross-spawn" from within the windows cmd.exe context. Calling quit/q/ctrl c closes the node process spawned by "ionic serve" and the cmd process spawned by cross spawn(running the gulp tasks), but the actual gulp process(gulp-watch) is not known in the context of ionic-app-lib/lib/serve.js.

I have verified this by comparing the pids of both node processes and the cmd.exe(spawned from cross-spawn) inside of serve.js(located in.

To Summarize:

Ionic serve spawns a nodejs and cmd.exe(when using gulp startup tasks) process
The cmd.exe process spawns the gulp tasks(specifically gulp-watch) inside a separate node process which becomes orphaned on exit(quit/q/Ctrl c).

System info:

Your system information:

Cordova CLI: 5.1.1
Gulp version:  CLI version 3.9.0
Gulp local:
Ionic CLI Version: 1.6.4
Ionic App Lib Version: 0.3.8
OS: Windows 7 SP1
Node Version: v0.12.7

**Comment by clmntfrndz** [comment]
Issue still present in ionic cli v1.7.11

**Comment by johnmcase** [comment]
+1 on this.  It is very frustrating having to remember to manually kill the node process every time I `q` out of `ionic serve`

My environment:

$ ionic info

Your system information:

Cordova CLI: 5.4.0
Gulp version:  CLI version 3.9.0
Gulp local:   Local version 3.9.0
Ionic Version: 1.1.1
Ionic CLI Version: 1.7.12
Ionic App Lib Version: 0.6.5
OS: Windows 7 SP1
Node Version: v4.2.2

**Comment by jacqueskang** [comment]
+1 I have the same issue. It says "Closing Gulp process" but it's not closing.

Your system information:

Cordova CLI: 5.4.1
Gulp version:  CLI version 3.9.0
Gulp local:   Local version 3.9.0
Ionic Version: 1.2.4
Ionic CLI Version: 1.7.14
Ionic App Lib Version: 0.7.0
OS:
Node Version: v4.3.1

**Comment by emilxp** [comment]
I have the same issue. I see that there has been a pull request for some time. Why isn't it already merged?

My environment:

Gulp version:  CLI version 1.2.1
Gulp local:   Local version 3.9.1
Ionic Framework Version: 1.2.4
Ionic CLI Version: 1.7.15
Ionic App Lib Version: 0.7.2
OS: Windows 7 SP1
Node Version: v0.12.2

**Comment by mlynch** [comment]
Seems to be a Windows issue? We're moving away from gulp so perhaps this will just fix itself.

**Comment by johnmcase** [comment]
Probably is a Windows issue.  Are you moving away from gulp in Ionic 1 as well?  Do ionic 1 & 2 have different CLIs?

I'm asking because I know there are going to be many companies that are going to be stuck on Ionic 1 for a long time due to not wanting to make the jump to Angular 2 and you should be supporting those folks as well.  I've been assured that Ionic 1 will continue to be supported for a long time.  If gulp is only going away in Ionic 2 but sticking around in Ionic 1 then there ought to be an actual fix.

**Comment by mlynch** [comment]
Yep, we will continue to support it, similarly to our support of mobile OSs and user adoption on them.

Can you try node 6 and possibly increment if the gulp version in package.json then doing an npm install? It might be that gulp fixed this and we're just defaulting to an old version

**Comment by jthoms1** [comment]
Thanks for opening the issue. It seems that issue has been resolved. I am going to close this issue but please feel free to reopen another issue if this occurs again.
